gslib/tests/test_mv.py

```diff
@@ -119,6 +119,61 @@ def test_move_dir_to_bucket(self):
     self.RunGsUtil(['mv', dir_to_move, suri(bucket_uri)])
     self.AssertNObjectsInBucket(bucket_uri, 2)
 
+  def test_move_bucket_objects_with_duplicate_names_to_bucket(self):
+    """Tests moving multiple top-level items between buckets."""
+    bucket1_uri = self.CreateBucket()
+    self.CreateObject(bucket_uri=bucket1_uri,
+                      object_name='dir1/file.txt',
+                      contents=b'data')
+    self.CreateObject(bucket_uri=bucket1_uri,
+                      object_name='dir2/file.txt',
+                      contents=b'data')
+    self.AssertNObjectsInBucket(bucket1_uri, 2)
+    bucket2_uri = self.CreateBucket()
+    self.AssertNObjectsInBucket(bucket2_uri, 0)
+
+    self.RunGsUtil(['mv', suri(bucket1_uri, '*'), suri(bucket2_uri)])
```

Comment:
If I understand correctly, this would work fine even without the change made for multiple top level sources given that the destination exists. I think we should check for a non-existing destination dir.

Reply:
I'm not sure it's the case that this would work before the PR.
This also copies to a non-existing destination dir (it copies to bucket).
But will specifically add a test copying to a non-existing bucket subdir.

Reply:
Bucket gets treated as an existing dir, since a bucket always exists if it is created.